ledger/db/src/ledger_trait.rs

```diff
@@ -90,4 +91,16 @@ pub trait Ledger: Send {
         }
         self.get_block(num_blocks - 1)
     }
+
+    /// Set active mint configurations for a given token id, based on a
+    /// set-mint-config transaction.
+    fn set_active_mint_configs(&self, set_mint_config_tx: &SetMintConfigTx) -> Result<(), Error>;
+
+    /// Get active mint configurations for a given token id.
+    /// Returns an empty array of no mint configurations are active for the
+    /// given token id.
+    fn get_active_mint_configs(&self, token_id: TokenId) -> Result<Vec<ActiveMintConfig>, Error>;
+
+    /// Update the total minted amount for a given MintConfig.
+    fn update_total_minted(&self, mint_config: &MintConfig, amount: u64) -> Result<(), Error>;
```

Comment:
This seems racy if there are multiple active minters. I think it would be more robust as a `log_minted` kind of method that increments the `total_minted`, which would be more retryable.

IIUC we're only planning on having one minter, so this might not be an issue in practice, but also the `log_minted` approach could enforce that the total is monotonically increasing.

Reply:
I don't believe this is racy since this the read-modify-write is done inside a single LMDB transaction.

Reply:
Just to clarify, the total minted amount is tracked per minting configuration. So there could be multiple `MintConfig`s currently configured, but the total minted amount is tracked individually for each one.

A followup PR might actually remove this from the public interface, since it can be done inside `append_block` - as the block has information about who minted what so there will not be a need for an external caller to call this.